File: app/src/main/java/com/example/kocja/rabbiter_online/adapters/BooleanConvertAdapter.java
package com.example.kocja.rabbiter_online.adapters;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;

import java.lang.reflect.Type;

public class BooleanConvertAdapter implements JsonDeserializer<Boolean> {
    @Override
    public Boolean deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        int code = json.getAsInt();

        return code != 0;
    }
}

File: app/src/main/java/com/example/kocja/rabbiter_online/adapters/BooleanConvertAdapter.kt
package com.example.kocja.rabbiter_online.adapters

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException

import java.lang.reflect.Type

class BooleanConvertAdapter : JsonDeserializer<Boolean> {
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Boolean? {
        val code = json.asInt

        return code != 0
    }
}
